cpp/src/arrow/status.h

```diff
@@ -368,27 +371,34 @@ class ARROW_EXPORT [[nodiscard]] Status : public util::EqualityComparable<Status
     StatusCode code;
     std::string msg;
     std::shared_ptr<StatusDetail> detail;
+    bool is_constant = false;
```

Comment:
Perhaps we can reduce the struct size by moving `is_constant` together with `code`.
```suggestion
    StatusCode code;
    bool is_constant = false;
    std::string msg;
    std::shared_ptr<StatusDetail> detail;
```